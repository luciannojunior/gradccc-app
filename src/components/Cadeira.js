import React, { Component } from 'react';
import { View, StyleSheet, Text, TouchableOpacity } from "react-native";
import { connect } from 'react-redux';

const styles = StyleSheet.create({
    container:{
        padding: 20,
        margin: 10,
        flexDirection: 'row',
        height: 70,
        backgroundColor: 'transparent',
        borderRadius: 4, borderWidth: 2, borderColor: '#d6d7da',
    },
    containerSelecionado: {
        padding: 20,
        margin: 10,
        flexDirection: 'row',
        height: 70,
        borderRadius: 4, borderWidth: 2, borderColor: '#fff',
        backgroundColor: '#069'
    },
    texto: {
      fontSize: 20
    },
    textoSelecionado: {
        fontSize: 20,
        color: '#fff'
    }
});

class Cadeira extends Component{

    componentWillMount() {
        this.setState({
            selecionado: false
        });
    };

    selecionou() {
        const { selecionado } = this.state;
        this.setState({selecionado: !selecionado});
    }

    render() {
        return (
        <TouchableOpacity onPress={this.selecionou.bind(this)}>
            <View style={(this.state.selecionado) ? styles.containerSelecionado : styles.container}>
                <Text style={(this.state.selecionado) ? styles.textoSelecionado : styles.texto}>{this.props.nomeCadeira}</Text>
            </View>
        </TouchableOpacity>);
    };
}

// Cadeira.prototype = {
//     nomeCadeira: React.PropTypes.string
// };

export default Cadeira;