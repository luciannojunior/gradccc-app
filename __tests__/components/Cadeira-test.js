import 'react-native';
import React from 'react';
import Cadeira from '../../src/components/Cadeira';

import renderer from 'react-test-renderer';

it('Cadeira selected renders correctly', () => {
  const tree = renderer.create(<Cadeira cadeira={selecionada=true}/>).toJSON();
  expect(tree).toMatchSnapshot();
});
it('Cadeira not selected renders correctly', () => {
    const tree = renderer.create(<Cadeira cadeira={selecionada=false}/>).toJSON();
    expect(tree).toMatchSnapshot();
});