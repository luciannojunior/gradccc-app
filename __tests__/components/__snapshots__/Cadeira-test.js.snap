// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cadeira not selected renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#d6d7da",
        "borderRadius": 4,
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 70,
        "margin": 10,
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;

exports[`Cadeira selected renders correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
        "borderColor": "#d6d7da",
        "borderRadius": 4,
        "borderWidth": 2,
        "flexDirection": "row",
        "height": 70,
        "margin": 10,
        "padding": 20,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      style={
        Object {
          "fontSize": 20,
        }
      }
    />
  </View>
</View>
`;
